src/cmd/linuxkit/util/reference_test.go

```diff
@@ -0,0 +1,59 @@
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReferenceExpand(t *testing.T) {
+	tests := []struct {
+		name    string
+		ref     string
+		options []ReferenceOption
+		want    string
+	}{
+		{
+			"basic image name should expand to docker.io/library image",
+			"redis",
+			nil,
+			"docker.io/library/redis",
+		},
+		{
+			"image name with user/org should expand to docker.io image",
+			"foo/bar",
+			nil,
+			"docker.io/foo/bar",
+		},
+		{
+			"custom registry image name should not expand",
+			"myregistry.io/foo",
+			nil,
+			"myregistry.io/foo",
+		},
+		{
+			"image name with more than three parts should not expand",
```

Comment:
Why should it not? Isn't this the equivalent of `docker.io/foo/bar/baz`? Since it has no host, isn't the default docker.io?

Reply:
The previous logic didn't expand in this scenario and a three part image wouldn't be a valid Docker Hub image afaik.

Reply:
> Your repository name must contain a combination of alphanumeric characters and may contain the special characters ., _, or -. Letters must be lowercase.

Docker Hub doesn't allow slashes in a name, so a three+ part repo is almost definitely not a Docker Hub repo.

Reply:
> The previous logic didn't expand in this scenario and a three part image wouldn't be a valid Docker Hub image afaik.

That is a valid point. Let's leave this, then, until someone in product management at Docker Hub decides to expand that. We can worry about it then.